src/main/java/com/aws/greengrass/mqttbridge/clients/IoTCoreClient.java

```diff
@@ -48,14 +62,36 @@ public class IoTCoreClient implements MessageClient {
         }
     };
 
+    private final MqttClientConnectionEvents connectionCallbacks = new MqttClientConnectionEvents() {
+        @Override
+        public void onConnectionInterrupted(int errorCode) {
+        }
+
+        @Override
+        public void onConnectionResumed(boolean sessionPresent) {
+            synchronized (subscribeFuture) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Object subscribeFuture synchronized on line 72 should not be re-assigned in the same synchronized block on line 79. After this re-assignment, you lose the reference to the synchronized object.